mmdet/models/backbones/darknet.py

```diff
@@ -0,0 +1,181 @@
+                            act_cfg=dict(type='LeakyReLU',
+                                         negative_slope=0.1)):
+    """In Darknet backbone, ConvLayer is usually followed by ResBlock. This
+    function will make that. The Conv layers always have 3x3 filters with
+    stride=2. The number of the filters in Conv layer is the same as the out
+    channels of the ResBlock.
+
+    Args:
+        in_channels (int): The number of input channels.
+        out_channels (int): The number of output channels.
+        res_repeat (int): The number of ResBlocks.
+        norm_cfg (dict): Dictionary to construct and config norm layer.
+            Default: dict(type='BN', requires_grad=True)
+        act_cfg (dict): Config dict for activation layer.
+            Default: dict(type='LeakyReLU', negative_slope=0.1).
+    """
+
+    cfg = dict(norm_cfg=norm_cfg, act_cfg=act_cfg)
+
+    model = nn.Sequential()
+    model.add_module(
+        'conv',
+        ConvModule(in_channels, out_channels, 3, stride=2, padding=1, **cfg))
+    for idx in range(res_repeat):
+        model.add_module('res{}'.format(idx), ResBlock(out_channels, **cfg))
+    return model
+
+
+@BACKBONES.register_module()
+class Darknet(nn.Module):
+    """Darknet backbone.
+
+    Args:
+        depth (int): Depth of Darknet. Currently only support 53.
+        out_indices (Sequence[int]): Output from which stages.
+        norm_cfg (dict): Dictionary to construct and config norm layer.
+            Default: dict(type='BN', requires_grad=True)
+        act_cfg (dict): Config dict for activation layer.
+            Default: dict(type='LeakyReLU', negative_slope=0.1).
+        norm_eval (bool): Whether to set norm layers to eval mode, namely,
+            freeze running stats (mean and var). Note: Effect on Batch Norm
+            and its variants only.
+
+    Example:
+        >>> from mmdet.models import Darknet
+        >>> import torch
+        >>> self = Darknet(depth=53)
+        >>> self.eval()
+        >>> inputs = torch.rand(1, 3, 416, 416)
+        >>> level_outputs = self.forward(inputs)
+        >>> for level_out in level_outputs:
+        ...     print(tuple(level_out.shape))
+        ...
+        (1, 256, 52, 52)
+        (1, 512, 26, 26)
+        (1, 1024, 13, 13)
+    """
+
+    def __init__(self,
+                 depth=53,
+                 out_indices=(3, 4, 5),
+                 norm_cfg=dict(type='BN', requires_grad=True),
+                 act_cfg=dict(type='LeakyReLU', negative_slope=0.1),
+                 norm_eval=True):
+        super(Darknet, self).__init__()
+        self.depth = depth
+        self.out_indices = out_indices
+        if self.depth == 53:
+            self.layers = [1, 2, 8, 8, 4]
+            self.channels = [[32, 64], [64, 128], [128, 256], [256, 512],
+                             [512, 1024]]
+        else:
+            raise KeyError(f'invalid depth {depth} for darknet')
```

Comment:
I suggest making this part an attribute of `Darknet` like what we did in `ResNet`.

Reply:
Sure. I'll put them into `arch_settings`